cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1340,7 +1332,8 @@ private void deleteApp(ApplicationId appId, ApplicationSpecification spec) throw
     deleteAppMetadata(appId, spec);
     store.deleteWorkflowStats(appId);
     deletePreferences(appId, spec);
-    store.removeApplication(appId);
+    store.removeApplicationData(appId.getAppReference());
```

Comment:
don't split this into two calls yet. Only do it if/when we're sure we need to make part of it async. Otherwise it's possible to successfully remove some of the data and then fail to remove the rest.

Reply:
got it